compiled_starters/grep-starter-go/cmd/mygrep/main.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	// Uncomment this to pass the first stage
+	// "bytes"
+	"fmt"
+	"io"
+	"os"
+)
+
+// Usage: echo <input_text> | your_grep.sh -E <pattern>
+func main() {
+	if len(os.Args) < 3 || os.Args[1] != "-E" {
+		fmt.Fprintf(os.Stderr, "usage: mygrep -E <pattern>\n")
+		os.Exit(2) // 1 means no lines were selected, >1 means error
+	}
+
+	pattern := os.Args[2]
+
+	line, err := io.ReadAll(os.Stdin) // suppose it is only one line
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error: read input text: %v\n", err)
+		os.Exit(2)
+	}
+
+	_, _ = pattern, line
```

Comment:
This line needed only to pass CI tests. Tester compiles starter code before uncommenting starter code, so `pattern` and `line` are unused in that case, and so Go compiler complains about it.
But when starter solution uncommented, both this variables are used and the code is compiled.

This limitation can be avoided if part "Uncomment this" is moved to a functions (which you suggested not to do), because unused function arguments are fine to the compiler.

I'll think, maybe I can come up with another smart solution to this.

Reply:
@nikandfor Nice work evaluating these options! 

I didn't think about the unused variables problem when I mentioned moving the uncomment marker to main. Having the `_, _` line is definitely more confusing than the uncomment marker being at the bottom of the file. The trade-off doesn't worth it. 

Feel free to use the function approach, that seems to be cleanest option among the lot. Just to make sure we're on the same page, I'm assuming the function will be relatively simple and look a bit like this (no loops etc.): 

```go
func match(line, pattern) (bool, err) {
  // You can log...
  print("whatever")
  
  // Uncomment this to pass
  // return containAny(line, pattern)
}
```